src/controllers/seller controller/sellerController.ts

```diff
@@ -52,99 +50,111 @@ interface PaginationResponse {
 }
 
 class SellerController {
-async getAllProductsBySeller(req: Request, res: Response): Promise<void> {
-  const token = req.cookies.jwt;
-  
-  if (!token) {
-    res.status(401).json({ message: 'Unauthorized: Token is missing' });
-    return;
-  }
-
-
-  try {
-    const decodedToken = jwt.verify(token, process.env.VERIFICATION_JWT_SECRET as string) as JwtPayload;
-    const userId = decodedToken.userId;
+  async getAllProductsBySeller(req: Request, res: Response): Promise<void> {
+    const token = req.cookies.jwt;
 
-    const userServiceInstance = new userService();
-    const user = await User.findByPk(userId)
-   console.log('my user is', userId)
-    if (!user) {
-      res.status(404).json({ message: 'User not found' });
-      return;
-    }
-    const userRole = await Role.findByPk(user.dataValues.roleId) as any;
-    if (userRole.dataValues.name !== 'seller') {
-      res.status(403).json({ message: 'Only sellers can access this resource' });
+    if (!token) {
+      res.status(401).json({ message: 'Unauthorized: Token is missing' });
       return;
     }
-    const productServiceInstance = new SellerService();
-    const products = await productServiceInstance.getProductsBySellerId(userId) as any;
-
-    // Pagination 
-    const page = req.query.page ? parseInt(req.query.page as string, 10) : 1;
-    const limit = req.query.limit ? parseInt(req.query.limit as string, 10) : 5;
-    const offset = (page - 1) * limit;
-    const totalProducts = products.length;
-
-    const paginatedProducts = products.map((product: Product): ProductWithSeller => {
-      const { productId, userId, productName, description, productCategory, price, quantity, images, dimensions, isAvailable, createdAt, updatedAt } = product;
-
-      return {
-        dataValues: product.dataValues,
-        productId,
-        userId,
-        productName,
-        description,
-        productCategory,
-        price,
-        quantity,
-        images,
-        dimensions,
-        isAvailable,
-        createdAt,
-        updatedAt,
-        seller: {
-          id: user.dataValues.id,
-          name: user.dataValues.firstname,
-          email: user.dataValues.email,
-        },
-      };
-    });
-
-    const productsWithSeller = paginatedProducts.slice(offset, offset + limit);
-
-    const response: PaginationResponse = {
-      message: 'Products fetched successfully',
-      products: productsWithSeller,
-      totalItems: totalProducts,
-      currentPage: page,
-      totalPages: Math.ceil(totalProducts / limit),
-    };
-
-    if (page < Math.ceil(totalProducts / limit)) {
-      response.next = {
-        page: page + 1,
-        limit,
-      };
-    }
-    
-    if (page > 1) {
-      response.previous = {
-        page: page - 1,
-        limit,
+
+    try {
+      const decodedToken = jwt.verify(token, process.env.VERIFICATION_JWT_SECRET as string) as JwtPayload;
```

Comment:
I think this is also un necessary as you are using a middleware before accessing this route